Kite SDK/PSPrintSDK/OLAnalytics.m

```diff
@@ -47,6 +47,132 @@ + (NSString *) platform{
     return platform;
 }
 
++ (NSString *)environment {
+    NSString *environment = @"Live";
+#ifdef PAYMENT_SANDBOX
+    environment = @"Development";
+#endif
+    return environment;
+}
+
++ (void)addPushDeviceToken:(NSData *)deviceToken {
+    const unsigned char *buffer = (const unsigned char *)[deviceToken bytes];
+    if (!buffer) {
+        return;
+    }
+    NSMutableString *hex = [NSMutableString stringWithCapacity:(deviceToken.length * 2)];
+    for (NSUInteger i = 0; i < deviceToken.length; i++) {
+        [hex appendString:[NSString stringWithFormat:@"%02lx", (unsigned long)buffer[i]]];
+    }
+    
+    NSString *pushToken = [NSString stringWithString:hex];
+    if (!pushToken) {
+        return;
+    }
+    
+    NSString *uuid = [self userDistinctId];
+    NSDictionary *properties = @{
+                                 @"uuid": uuid,
+                                 @"set" : @{
+                                         @"push_token" : @{
+                                                 @"platform" : @"iOS",
+                                                 @"token" : pushToken
+                                                 },
+                                         @"platform" : @"iOS",
+                                         @"environment": [self environment]
+                                         }
+                                 };
+    
+    AFHTTPRequestOperationManager *manager = [AFHTTPRequestOperationManager manager];
+    manager.requestSerializer = [AFJSONRequestSerializer serializer];
+    [manager.requestSerializer setValue:[NSString stringWithFormat:@"apikey %@", [OLKitePrintSDK apiKey]] forHTTPHeaderField:@"Authorization"];
+    [manager.requestSerializer setValue:@"application/json" forHTTPHeaderField:@"Content-Type"];
+    [manager POST:[NSString stringWithFormat:@"%@/v1.4/person/", [OLKitePrintSDK apiEndpoint]] parameters:properties success:^(AFHTTPRequestOperation *operation, id responseObject) {
+        if ([operation.response statusCode] >= 200 && [operation.response statusCode] <= 299){
+            NSLog(@"Successfully posted push notification token.");
+        }
+        else{
+            NSLog(@"There was an error posting the push notification token: %ld", (long)[operation.response statusCode]);
+        }
+        
+    }failure:^(AFHTTPRequestOperation *operation, NSError *error) {
+        NSLog(@"There was an error posting the push notification token: %@", error);
+    }];
+}
+
++ (void)trackReceivedPushNotification:(NSDictionary *)userInfo{
+    NSDictionary *dict = [OLAnalytics defaultDictionaryForEventName:@"Push Received"];
+    [dict[@"properties"] setObject:userInfo forKey:@"push_userInfo"];
+    [OLAnalytics sendToMixPanelWithDictionary:dict];
```

Comment:
@kkarayannis Can you pick up with @drakethomas33 with regards to a server endpoint to record this push notification open "conversion" against this user as we want to be able to display notification -> open conversion rate within Kite for individual customers based on their X-Kite-UUID